src/chmatch.c

```diff
@@ -1,7 +1,13 @@
 #include "data.table.h"
 
 static SEXP chmatchMain(SEXP x, SEXP table, int nomatch, bool chin, bool chmatchdup) {
-  if (!isString(x) && !isNull(x)) error(_("x is type '%s' (must be 'character' or NULL)"), type2char(TYPEOF(x)));
+  if (!isString(x) && !isNull(x)) {
+    // for use in sub_*_funs from nse_utils.R
+    if (TYPEOF(x) == SYMSXP) {
+      return chmatchMain(coerceVector(x, STRSXP), table, nomatch, chin, chmatchdup);
```

Comment:
Do we need a `PROTECT()` here?

Reply:
We don't need it here but in the function that will this this one

Reply:
We need